Task 1/Assets/Scripts/GameController.cs

```diff
@@ -0,0 +1,77 @@
+using UnityEngine;
+
+public class GameController : MonoBehaviour
+{
+    public Vector3 initialSize;
+    public Vector3 initialPosition;
+    public Vector3 startingBoxPosition;
+    public Vector3 startMovingPosition = new Vector3(-1.5f, 0, 0);
+
+    private GameLogic logic;
+    private Camera cam;
+    private Vector3 cameraStartPos;
+    public int Score { get; private set; }
+    public int MaxScore { get; private set; }
+
+    private bool isStarting;
+
+    private void Start()
+    {
+        logic = GetComponent<GameLogic>();
+        logic.Init(initialSize,initialPosition);
+
+        var posY = Mathf.Abs(startingBoxPosition.y*2 + initialSize.y / 2);
+        logic.GenerateStartingBox(new Vector3(initialSize.x, posY,initialSize.z),startingBoxPosition);
+
+        Score = logic.Count;
+        EventManager.GameStarting += OnGameStarting;
+        EventManager.PlaceSuccess += OnPlaceSuccess;
+        EventManager.GameEnding += OnGameEnding;
+        cam = Camera.main;
+        cameraStartPos = cam.transform.position;
+    }
+
+    private void OnGameEnding()
+    {
+        if (Score>MaxScore)
```

Comment:
hint: you can use Mathf.Max:
```suggestion
       MaxScore = Mathf.Max(Score, MaxScore);
```